src/wp-includes/comment.php

```diff
@@ -1754,6 +1754,40 @@ function wp_get_current_commenter() {
 	return apply_filters( 'wp_get_current_commenter', compact('comment_author', 'comment_author_email', 'comment_author_url') );
 }
 
+/**
+ * Get unapproved comment author's email.
+ *
+ * Used to allow the commenter to see their pending comment.
+ *
+ * @since WP-5.1.0
+ *
+ * @return string The unapproved comment author's email (when supplied).
+ */
+function wp_get_unapproved_comment_author_email() {
```

Comment:
This also needs to be done using a full backport of the changeset(s) that added this function.

Reply:
Actually that's not quite right either. `wp_get_unapproved_comment_author_email` was added as part of a larger change: https://core.trac.wordpress.org/changeset/44659 / https://core.trac.wordpress.org/ticket/43857

If that is adopted then we would also need to backport https://core.trac.wordpress.org/ticket/49956 which is a bugfix for that change.

What should be done here is to make the change desired in the PR, but without using the `wp_get_unapproved_comment_author_email` function. The logic should look like what was present before that function was added: https://core.trac.wordpress.org/changeset/44659#file1